include/core/Partition_One.hpp

```diff
@@ -0,0 +1,40 @@
+#ifndef PARTITION_ONE_H
+#define PARTITION_ONE_H
+
+#ifdef NGEN_MPI_ACTIVE
```

Comment:
The guard for `NGEN_MPI_ACTIVE` isn't necessary here. MPI's not actually referenced or used by any of this code. Removing the guard would simplify test code by removing the weird `#define NGEN_MPI_ACTIVE`

Reply:
The presence of ```NGEN_MPI_ACTIVE``` does make the code look a bit cumbersome. However, Partition_One is specifically designed for this case. The source codes are under this condition. On the other hand, I assume if the test code works w/o MPI. it is probably ok.

Reply:
The ```NGEN_MPI_ACTIVE``` is needed as Partition_One.hpp needs it. Without it, the test code won't compile. The Parttion_Test.cpp using Partition_Parser.hpp works in a similar fashion. So this is kept in place.